### ISSUE TITLE
[Feature] YOLO loss definition

### ISSUE THREAD

**Issue by rehohoho** [whizzmobility/models#38]
# Prerequisites

Please answer the following question for yourself before submitting an issue.

- [x] I checked to make sure that this feature has not been requested already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/official/vision/beta

## 2. Describe the feature you request

YOLO loss definition consists of 
- generalised IoU (considers shape, orientation in addition to overlap)
- center IoU (considers overlap, distance between center, aspect ratio)
- distance IoU (considers distance of center of bbox)

## 3. Additional context

Add any other context about the feature request here.

## 4. Are you willing to contribute it? (Yes or No)

Yes

**Comment by rehohoho** [comment]
https://github.com/miemie2013/Keras-YOLOv4/tree/master/model
https://github.com/wangermeng2021/Scaled-YOLOv4-tensorflow2/blob/main/model/losses.py

Check against other implementations. The one that was modelled after seems fishy..

**Comment by rehohoho** [comment]
Closing this for now, as the model converges properly and infers well. Open a new issue if any component of loss seems to be dubious.
